pkg/database/key_server_stats.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm/dialects/postgres"
+	"github.com/lib/pq"
+)
+
+// KeyServerStats represents statistics for a key-server for this realm
+type KeyServerStats struct {
+	Errorable
+
+	// RealmId that these stats belong to.
+	RealmID uint `gorm:"column:realm_id; primary_key; type:integer; not null;"`
+
+	// IsSystem determines if this is a system-level stats configuration. There can
+	// only be one system-level stats configuration.
+	IsSystem bool `gorm:"column:is_system; type:bool; not null; default:false;"`
+
+	// KeyServerURL allows a realm to override the system's URL with its own
+	KeyServerURL string `gorm:"column:key_server_url; type:varchar(150); default: '';"`
+	// KeyServerAudience allows a realm to override the system's audience
+	KeyServerAudience string `gorm:"column:key_server_audience; type:varchar(150); default: '';"`
+
+	Stats []*KeyServerStatsDay `gorm:"PRELOAD:false; SAVE_ASSOCIATIONS:false; ASSOCIATION_AUTOUPDATE:false, ASSOCIATION_SAVE_REFERENCE:false;"`
```

Comment:
Don't do this, gorm will make you have a bad day. The way we've done this in other models is to have a public method on the parent struct that accepts a database.Database as the first parameter and runs the query.

```go
func (s *KeyServerStats) DailyStats(db *Database) ... {
}
```